Competition2017/src/competition/subsystems/drive/commands/AutonomousDriveAlongCurveCommand.java

```diff
@@ -11,69 +11,79 @@
 import xbot.common.properties.DoubleProperty;
 import xbot.common.properties.XPropertyManager;
 
-public class DriveAlongCurveCommand extends BaseDriveCommand {
+public class AutonomousDriveAlongCurveCommand extends BaseDriveCommand {
     
-    private final DoubleProperty totalTime;
+    private final DoubleProperty totalTimeOfAutoCurve;
     private final DoubleProperty timeStep;
     private final DoubleProperty robotTrackWidth;
     
-    private final double initialCommandStart;
-    private int i;
+    private double[][] waypoints;
     
-    AutonomousCurveTrajectoryPlanner path;
+    private final double initialCommandTime;
+    private int indexOfTimeStep;
+    
+    final AutonomousCurveTrajectoryPlanner path;
     
     @Inject
-    public DriveAlongCurveCommand
+    public AutonomousDriveAlongCurveCommand
             (DriveSubsystem driveSubsystem,
             XPropertyManager prop
             ) {
         
         super(driveSubsystem);
         this.requires(driveSubsystem);
         
-        this.i = 0;
-        this.initialCommandStart = Timer.getFPGATimestamp();
+        this.indexOfTimeStep = 0;
+        this.initialCommandTime = Timer.getFPGATimestamp();
         
         //Input points to create trajectory for robot
-        double[][] waypoints = new double[][]{
+         this.waypoints = new double[][]{
```

Comment:
If no path is specified, it'd be safer to have the robot do nothing (i.e., have no points here) than to have it move without our control.